### DISCUSSION TITLE
Layout Component destroyed my code (hopefully not)

### DISCUSSION THREAD

**Discussion by ostermann-felix** [neuefische/questions#178] [Web] [answered]
Hi friends, today I tried the layout component but I think I missed something. Now the page looks like this: 

I don't get any error, it just don't show anything else that the header...

<img width="1411" alt="Bildschirmfoto 2022-09-28 um 13 57 57" src="https://user-images.githubusercontent.com/94837451/192782103-439217d6-075c-4ea9-8813-d30d13d98f65.png">

The code I wrote is here:

Layout Component
![Bildschirmfoto 2022-09-28 um 13 58 05](https://user-images.githubusercontent.com/94837451/192782151-5d400a9e-e20c-4547-b496-06ec84f27a57.png)

Header Component 
![Bildschirmfoto 2022-09-28 um 13 58 22](https://user-images.githubusercontent.com/94837451/192782313-b653a907-47ba-482a-9eb8-7627d5c1c2a4.png)

Footer Component
![Bildschirmfoto 2022-09-28 um 13 58 35](https://user-images.githubusercontent.com/94837451/192782363-dd66436f-9bc8-4d55-a3d3-a969dbc36654.png)

**Comment by NikRoe** [comment]
Hey Felix,

can you send us a PR? 

Thanks in advance

**Reply by ostermann-felix** [reply]
https://github.com/ostermann-felix/capstone-project/pull/9

**Reply by NikRoe** [reply]
Thanks for the PR!

It looks like your content is simply hidden by your header component.
When assigning the property `position: fixed;` you can also determine where the element shall be fixed (for instance `top: 0;`) [position on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/position). But assigning the property `position: fixed;` will remove the element from the normal flow of the document, thus leading to the overlap you currently have.

You should then take a closer look into your overall CSS as to how you would want to restructure it so that there is no longer an overlap.

Hope this already helps 🤞

**Reply by ostermann-felix** [reply]
I will give it a try.

**Reply by ostermann-felix** [reply]
Okay I don't have it! What in perticular do you mean?

**Comment by CJAkkuec** [comment, marked as answer]
Hey there, I took a quick look and there's some things that could be optimized. It's not really so much the Layout component that is broken, but some of the CSS rules.

I've also noticed your Fast Refresh is busted, but I'm not sure why. 🤷‍♀️ 

**Most importantly**
`GlobalStyle.js`
If you declare the `max-width `of your `body` tag to be `375px `in your `GlobalStyle.js`, but your `Header` and `Footer` components are both 100vw long, you'll quickly run into issues. You can set your `Header` and `Footer` width to 100% instead.

You may want to adjust your universal selector to look like this and include the following rules:

```
  *,
  *::before,
  *::after {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Roboto', sans-serif;
  }
```

`Layout.js`
You may want to remove the additional `div` tag and just use fragments in its place.

Like so:
```
 <>
      <Header />
      {children}
      <Footer />
</>
```

Now, the reason why you're not seeing your content is because you haven't made it visible. It's being rendered, but it's stuck behind your header, never to be seen.

What you may want to include is a Wrapper or Container component that will put some space around your content.

Your StyledContainer.js component could look like this:

```
import styled from 'styled-components';

const StyledContainer = styled.div`
  margin: 3rem auto;
  padding: 1rem;
`;

export default StyledContainer;
```
Once you've got that sorted out, import your StyledContainer component into your Layout.js, and it should look like this:

```
import Header from './Header';
import Footer from './Footer';
import StyledContainer from './styledContainer';

const Layout = ({ children }) => {
  return (
    <>
      <Header />
      <StyledContainer>{children}</StyledContainer>
      <Footer />
    </>
  );
};

export default Layout;
```

If this doesn't work, kill your node and restart your server, because something about your refresh is busted that I can't figure out right now. It may be related to how you export your functions inside your components. _Buuuut_ the above will at least fix your layout. Kind of.

If you're wondering why your `Footer` isn't being displayed: it doesn't have a height. If you put any height rule into your `Footerbody` component, it will appear. Something's not quite right with your icons, I'm afraid.

**Comment by ostermann-felix** [comment]
Solved!

**Reply by CJAkkuec** [reply]
If you don't mind, could you perhaps offer some insight into how you solved your issue? :) When someone comes across your question at a later point, they might be interested.

**Reply by ostermann-felix** [reply]
Problems:
#1 All elements where overlayed at the top so that just the header was shown.
#2 The elements where not styled properly and bounded at the right top corner of the app.

Answers: 
#1 A<Main> wrapper around the {children} prop and a styling of this <Main> tag was the main issue solver.
#2 Position fixed with top, left and right orders fixed the spacing between header and pageside.
